docs/bsp.md

```diff
@@ -830,10 +830,25 @@ trait TaskFinishParams {
     /** Optional metadata about the task result.
       * For compilation tasks, this should be a CompileReport object.
       * For test tasks, this should be a TestReport object. */
-    def data: Option[Json]
+    def report: Option[TaskData]
```

Comment:
How about keeping `data: Option[Json]` unchanged and inlining `kind: Option[String]`? That avoids `TaskData` altogether
```scala
  def dataKind: Option[String]
    /** Optional metadata about the task result.
      * For compilation tasks, this should be a CompileReport object.
      * For test tasks, this should be a TestReport object. */
    def data: Option[Json]
}
object TaskDataKind {
  val CompileReport = "compile-report"
  // ...
} 
```

Reply:
I think I like this better afterall, yes. But maybe with one less optional:
```
/** The kind of the `data` field. "NONE" if `data` is empty. */
def dataKind: String
def data: Option[Json]

object DataKind {
  val NONE = "NONE"
  val CompileReport = "compile-report"
  // ...
}
```

Reply:
I think we should use what makes more sense in JSON which is that the `dataKind` key is either null or undefined, in bsp4j it will be a `String` either way.

Reply:
It's `Option[String]` in bsp4s to support the case when the key is missing, I think we actually have too many non-optional fields

Reply:
What do you mean? When it's not optional on the porotocol level it's simply a protocol error when it's not there, for optional it's up to the client to figure things out.

But I guess in this case an option is just as well.